providers/jldap/src/java/edu/amc/sakai/user/JLDAPDirectoryProvider.java

```diff
@@ -616,7 +616,7 @@ public boolean getUserbyAid(String aid, UserEdit user)
 	}
 
 	public LdapUserData getUserByAid(String aid, LDAPConnection conn) {
-		String filter = ldapAttributeMapper.getFindUserByAidFilter(aid);
+		String filter = ldapAttributeMapper.getFindUserByAidFilter(aid.replaceAll("@OX.AC.UK", ""));
```

Comment:
Robert mentioned that in the shibboleth the domain is lowercase and so it should probably be`@ox.ac.uk`.

`replaceAll()` takes a regular expressions and should be escaped.

Reply:
Ok, fixed.